tests/run_ziti.py

```diff
@@ -0,0 +1,185 @@
+#  Copyright (c) 2022.  NetFoundry, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#!/usr/bin/env python3
+import json, os, time, subprocess, platform, shutil, stat, sys, tarfile, tempfile, urllib.request as url
+
+if sys.platform == "win32":
+    sys.stdout.reconfigure(encoding="utf-8")
+
+GITHUB_REPO_OWNER = os.getenv("GITHUB_REPO_OWNER", "openziti")
+GITHUB_REPO_NAME  = os.getenv("GITHUB_REPO_NAME",  "ziti")
+GITHUB_TOKEN      = os.getenv("GITHUB_TOKEN", "")
+
+def _arch():
+    m = platform.machine().lower()
+    if m in ("x86_64", "amd64"):         return "amd64"
+    if m in ("aarch64", "arm64"):        return "arm64"
+    if m.startswith("arm"):              return "arm"
+    raise RuntimeError(f"unsupported arch: {m}")
+
+def _os():
+    p = sys.platform
+    if   p.startswith("linux"):   return "linux"
+    elif p == "darwin":           return "darwin"
+    elif p in ("win32", "cygwin"):return "windows"
+    raise RuntimeError(f"unsupported os: {p}")
+
+def _github_json(path):
+    req = url.Request(f"https://api.github.com{path}")
+    if GITHUB_TOKEN: req.add_header("Authorization", f"Bearer {GITHUB_TOKEN}")

```

Comment:
is it needed if it works without it?

Reply:
the github runners were routinely indicating the runtime hit a rate limit :/